### PR TITLE
SvgColourMatrixFix fix

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### What does this implement/fix? Explain your changes.

Fixes string Split for matrix type.

#### Repro:

```svg
<svg width="100%" height="100%" viewBox="0 0 150 360"
    preserveAspectRatio="xMidYMid meet"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink">

  <!-- ref -->
  <defs>
    <g id="circles">
      <circle cx="30" cy="30" r="20" fill="blue" fill-opacity="0.5" />
      <circle cx="20" cy="50" r="20" fill="green" fill-opacity="0.5" />
      <circle cx="40" cy="50" r="20" fill="red" fill-opacity="0.5" />
    </g>
  </defs>
  <use href="#circles" />
  <text x="70" y="50">Reference</text>

  <!-- matrix -->
  <filter id="colorMeMatrix">
    <feColorMatrix in="SourceGraphic"
        type="matrix"
        values="0 0 0 0 0
                1 1 1 1 0
                0 0 0 0 0
                0 0 0 1 0" />
  </filter>
  <use href="#circles" transform="translate(0 70)" filter="url(#colorMeMatrix)" />
  <text x="70" y="120">matrix</text>

  <!-- saturate -->
  <filter id="colorMeSaturate">
    <feColorMatrix in="SourceGraphic"
        type="saturate"
        values="0.2" />
  </filter>
  <use href="#circles" transform="translate(0 140)" filter="url(#colorMeSaturate)" />
  <text x="70" y="190">saturate</text>

  <!-- hueRotate -->
  <filter id="colorMeHueRotate">
    <feColorMatrix in="SourceGraphic"
        type="hueRotate"
        values="180" />
  </filter>
  <use href="#circles" transform="translate(0 210)" filter="url(#colorMeHueRotate)" />
  <text x="70" y="260">hueRotate</text>

  <!-- luminanceToAlpha -->
  <filter id="colorMeLTA">
    <feColorMatrix in="SourceGraphic"
        type="luminanceToAlpha" />
  </filter>
  <use href="#circles" transform="translate(0 280)" filter="url(#colorMeLTA)" />
  <text x="70" y="320">luminanceToAlpha</text>
</svg>
```

### CODE REVIEW

File: Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs [hunk 1]
```diff
@@ -87,7 +87,7 @@ public override void Process(ImageBuffer buffer)
                     };
                     break;
                 default: // Matrix
-                    var parts = this.Values.Replace("  ", " ").Split(new char[] { ' ', '\t', '\n', '\r', ',' });
+                    var parts = this.Values.Replace("  ", " ").Split(new char[] { ' ', '\t', '\n', '\r', ',' }, StringSplitOptions.RemoveEmptyEntries);
```
Comment (H1Gdev) [inline Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: I think replacement is also unnecessary.
```cs
var parts = this.Values.Split(new char[] { ' ', '\t', '\n', '\r', ',' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment (wieslawsoltes) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: Yes, pushed change

File: Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs [hunk 2]
```diff
@@ -87,7 +87,7 @@ public override void Process(ImageBuffer buffer)
                     };
                     break;
                 default: // Matrix
-                    var parts = this.Values.Replace("  ", " ").Split(new char[] { ' ', '\t', '\n', '\r', ',' });
+                    var parts = this.Values.Split(new char[] { ' ', '\t', '\n', '\r', ',' }, StringSplitOptions.RemoveEmptyEntries);
                     colorMatrixElements = new float[5][];
```
Comment (mrbean-bremen) [inline Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: Thanks - good catch! Could you add your test svg together with the corresponsing png as a regression test?

Comment (wieslawsoltes) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: I can do this but later if that's ok?

Comment (mrbean-bremen) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: Sure!

Comment (wieslawsoltes) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: @mrbean-bremen Seems like the png is not rendered correctly using SVG lib. The issue was only related to the string parsing not rendering.

Comment (wieslawsoltes) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: SVGViewer_2019-08-04_22-03-26

Comment (wieslawsoltes) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: The output from my renderer:
element-feColorMatrix

You can see one issue, the `Filter` property is inherited by Children of the `use` element, which is not what spec says. The `Fileter` property is not inheritable: https://www.w3.org/TR/SVG11/filters.html#FilterProperty, the same issue I had was with `Opacity`: https://www.w3.org/TR/SVG11/masking.html#ObjectAndGroupOpacityProperties

Comment (mrbean-bremen) [reply Source/Filter Effects/feColourMatrix/SvgColourMatrix.cs]: Thanks for digging into this! I will merge as is in this case. Filters are generally very poorly supported (see #43), so I'm not too surprised by this, though the inheritance problem is something that certainly shall be fixed.

**Pull request merged**